pymeasure/instruments/hp/hp3437A.py

```diff
@@ -0,0 +1,412 @@
+    def _decode_status(self, status_bytes, field=None):
+        """Method to decode the status bytes
+
+        :param status_bytes: list of bytes to be decoded
+        :param field: name of field to be returned
+        :return ret_val: int status value
+
+        """
+        ret_val = self.Status(self.status_bytes(*status_bytes))
+        if field is None:
+            return ret_val.b
+
+        if field == "SRQ":
+            return self.SRQ(getattr(ret_val.b, field))
+
+        if field == "Number":
+            bcd_nr = struct.pack(">I", getattr(ret_val.b, field))
+            return self._convert_from_bcd(bcd_nr)
+
+        if field == "Delay":
+            bcd_delay = struct.pack(">I", getattr(ret_val.b, field))
+            delay_value = (
+                self._convert_from_bcd(bcd_delay) / 1.0e7
+            )  # delay resolution is 100ns
+            return delay_value
+        return getattr(ret_val.b, field)
+
+    @staticmethod
+    def _decode_range(self, status_bytes):
+        """Method to decode current range
+
+        :param range_undecoded: int to be decoded
+        :return cur_range: float value repesenting the active measurment range
+        :rtype cur_range: float
+
+        """
+        cur_stat = self.Status(self.status_bytes(*status_bytes))
+        range_undecoded = cur_stat.b.range
+        # range decoding
+        # (cf table 3-2, page 3-5 of the manual, HPAK document 9018-05946)
+        if range_undecoded == 0:
+            cur_range = math.nan
+        # 1 indicates 0.1V range
+        if range_undecoded == 1:
+            cur_range = 0.1
+        # 2 indicates 10V range
+        if range_undecoded == 2:
+            cur_range = 10.0
+        # 3 indicates 1V range
+        if range_undecoded == 3:
+            cur_range = 1.0
+        return cur_range
+
+    @staticmethod
+    def _decode_trigger(self, status_bytes):
+        """Method to decode trigger mode
+
+        :param status_bytes: list of bytes to be decoded
+        :return trigger_mode: string with the current trigger mode
+        :rtype trigger_mode: str
+
+        """
+        cur_stat = self.Status(self.status_bytes(*status_bytes))
+        trig = cur_stat.b.trigger
+        if trig == 0:
+            log.error("HP3437A invalid trigger detected!")
+            trigger_mode = "INVALID"
+        if trig == 1:
+            trigger_mode = "Internal"
+        if trig == 2:
+            trigger_mode = "external"
+        if trig == 3:
+            trigger_mode = "hold/manual"
+        return trigger_mode
+
+    @staticmethod
+    def _unpack_data(self, data):
+        """
+        Method to unpack the data from the returned bytes in packed mode
+
+        :param data: list of bytes to be decoded
+        :return ret_data: float value
+
+        """
+        ret_data = self.packed_data(self.packed_bytes(*data))
+        return float(ret_data.b)
+
+    # commands overwriting the base implementaiton
+    def read(self):
+        """
```

Comment:
I think, this method is used to read measurements from instrument, so I would think it is a bit misleading to call it `read` as the method with same name in the parent class is meant to read generic response from instrument.
My proposal would be either to rename the method to something like `read_measurement` (or similar) or have an `Instrument.measurement` property that would simply read from instrument (I am not sure the latter proposal is technically feasible, though)

Reply:
My Intention of overwriting the `read` method was to catch two problems:

1. the problem of data decoding if the packed data format is selected
2. find a way to handle the GPIB-timeout related problems if a high number of readings and a long delay is set. Which I still need to debug further


I could of course rename the function to `read_measurement`, but then we should aslso add the warning that the `read` method may not correctly deliver data